dftimewolf/lib/state.py

```diff
@@ -208,6 +209,30 @@ def RunModules(self):
     """Performs the actual processing for each module in the module pool."""
     self._InvokeModulesInThreads(self._RunModuleThread)
 
+  def RegisterStreamingCallback(self, target, container):
+    """Registers a callback for a type of container.
+
+    Args:
+      target (function): function to be called.
+      container (interface.AttributeContainer): container type on which
```

Comment:
It's actually the container type that's relevant here, so "container whose type..."

But maybe this method should just take the type instead?

Reply:
We're technically passing a type (e.g. `container.Report`). It just happens that AttributeContainers have an attribute that describes their type as strings. I changed this to use the actual types instead of the attribute (see f5f8f14)